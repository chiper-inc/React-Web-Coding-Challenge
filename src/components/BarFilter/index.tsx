import { useState } from 'react';
import './style.scss';

const BarFilter = () => {
  return (
    <div className="bar-filter">
      <div className="content-search w-full md:w-3/12">
        <input
          className="input-search"
          placeholder="Search case descriptions"
        />
      </div>
      <div className="content-range-date"></div>
    </div>
  );
};

export default BarFilter;
