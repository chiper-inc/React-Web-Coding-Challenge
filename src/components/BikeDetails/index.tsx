import { useNavigate } from "react-router-dom";
import { useFormatDate } from "../../hooks/useFormatDate";
import Bike from "../../assets/images/bike.svg";
import { Wrapper, ContainerImg, ContainerInfo, Img, Row } from "./styles";

interface BikeDetailsProps {
  info: {
    id: number;
    thumb: string;
    title: string;
    description: string;
    date_stolen: number;
    stolen_location: string;
    frame_colors: string[];
    frame_material_slug: string;
    frame_model: string;
    frame_size: string;
    manufacturer_name: string;
    serial: string;
    type_of_cycle: string;
    year: number;
  }
}

export const BikeDetails = ({ info: {
  id,
  thumb,
  title,
  description,
  date_stolen,
  stolen_location,
  frame_colors,
  frame_material_slug,
  frame_model,
  frame_size,
  manufacturer_name,
  serial,
  type_of_cycle,
  year
}}: BikeDetailsProps) => {

  const navigate = useNavigate();
  const [date] = useFormatDate(date_stolen);

  const handleNavigate = () => navigate(`/case/${id}`);

  return (
    <Wrapper onClick={handleNavigate} >
      <ContainerImg>
        <Img src={thumb || Bike} alt={title} />
      </ContainerImg>
      <ContainerInfo>
        <span><b>Description of incident</b></span>
        <span>{description || 'No description.'}</span><br/>
        <Row>
          <span><b>Date Stolen:</b> {date || 'Unknown.'}</span>
          <span><b>Stolen Location:</b> {stolen_location || 'Unknown.'}</span>
        </Row>
        <Row>
          <span><b>Color(s):</b> {frame_colors.join(', ') || 'Unknown.'}</span>
          <span><b>Material:</b> {frame_material_slug || 'Unknown.'}</span>
        </Row>
        <Row>
          <span><b>Manufacturer:</b> {manufacturer_name || 'Unknown.'}</span>
          <span><b>Model:</b> {frame_model || 'Unknown.'}</span>
        </Row>
        <Row>
          <span><b>Serial:</b> {serial || 'Unknown.'}</span>
          <span><b>Size:</b> {frame_size?.toUpperCase() || 'Unknown.'}</span>
        </Row>
        <Row>
          <span><b>Type:</b> {type_of_cycle || 'Unknown.'}</span>
          <span><b>Year:</b> {year || 'Unknown.'}</span>
        </Row>
      </ContainerInfo>
    </Wrapper>
  );
};
