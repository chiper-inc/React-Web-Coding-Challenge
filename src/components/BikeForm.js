import React, { useContext, useState } from "react";
import { Bike } from "../context/bike";

const BikeForm = () => {
  const { dispatch } = useContext(Bike);

  const [title, setTitle] = useState("");
  const [from, setFrom] = useState("");
  const [to, setTo] = useState("");

  function handleTitleChange(e) {
    setTitle(e.target.value);
    dispatch({ key: ['search', 'title'], payload: e.target.value });
    dispatch({ type: 'FILTER' });
  }

  function handleFromChange(e) {
    setFrom(e.target.value);
    dispatch({ key: ['search', 'from'], payload: new Date(e.target.value).getTime() });
    dispatch({ type: 'FILTER' });
  }

  function handleToChange(e) {
    setTo(e.target.value);
    dispatch({ key: ['search', 'to'], payload: new Date(e.target.value).getTime() });
    dispatch({ type: 'FILTER' });
  }

  function handleClear(e) {
    setTitle('');
    setFrom('');
    setTo('');
    dispatch({ type: 'CLEAR' });
    dispatch({ type: 'FILTER' });
  }


  return (<>
    <Form
      title={title}
      from={from}
      to={to}
      handleTitleChange={handleTitleChange}
      handleFromChange={handleFromChange}
      handleToChange={handleToChange}
      handleClear={handleClear}
    />
  </>);
};


const Form = ({ title, from, to, handleTitleChange, handleFromChange, handleToChange, handleClear }) =>
  <div className="row">
    <div className="col-md-12">
      <br />
      <div className="input-group">
        <input
          data-testid="input-data-title"
          className="form-control"
          value={title}
          autoFocus={true}
          placeholder="Title"
          onChange={handleTitleChange}
        />

        <input
          data-testid="input-data-from"
          type="date"
          className="form-control"
          value={from}
          autoFocus={true}
          placeholder="From"
          onChange={handleFromChange}
        />
        <input
          data-testid="input-data-to"
          type="date"
          className="form-control"
          value={to}
          autoFocus={true}
          placeholder="From"
          onChange={handleToChange}
        />
        <div className="input-group-append">
          <button data-testid="btn-clear" className="btn btn-primary" onClick={handleClear}>
            Clear
          </button>
        </div>
      </div>
    </div>
  </div>;

export default BikeForm;
