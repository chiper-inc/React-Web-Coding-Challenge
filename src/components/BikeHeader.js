import React from "react";

export default () => (
  <div className="row justify-content-center">
    <div className="col-md-8">
      <h2>Police of departament of Berlin</h2>
      <h4>Stolen bykes</h4>
    </div>
  </div>
);
