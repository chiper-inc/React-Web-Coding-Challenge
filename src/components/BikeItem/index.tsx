import { useNavigate } from "react-router-dom";
import { useFormatDate } from "../../hooks/useFormatDate";
import bike from "../../assets/images/bike.svg";
import { Wrapper, ContainerInfo, ContainerImg, Description } from "./styles";

interface IBikeItemProps {
  id: number;
  thumb: string;
  title: string;
  description: string;
  date_stolen: number;
  stolen_location: string;
}

export const BikeItem = ({
  id, thumb, title, description, date_stolen, stolen_location
}: IBikeItemProps) => {

  const navigate = useNavigate();
  const [date] = useFormatDate(date_stolen);

  const handleNavigate = () => navigate(`/case/${id}`);

  return (
    <Wrapper onClick={handleNavigate} >
      <ContainerImg>
        <img className="bike-img" src={thumb || bike} alt={title} />
      </ContainerImg>
      <ContainerInfo>
        <span><b>Title:</b> {title || 'No title'}</span>
        <Description><b>Description:</b> {description || 'No description'}</Description>
        <span><b>Date Stolen:</b> {date || 'Unknown'}</span>
        <span><b>Stolen Location:</b> {stolen_location || 'Unknown'}</span>
      </ContainerInfo>
    </Wrapper>
  );
};
