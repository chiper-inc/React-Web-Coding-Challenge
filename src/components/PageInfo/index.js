import React from 'react'
import { Container } from './styles'
import icon from './assets/austronat.svg'

export const PageInfo = ({ children }) => (
  <Container className='container row'>
    <div className='col-md-6 align-self-center'>
      <img src={icon} />
    </div>
    <div className='col-md-6 align-self-center'>{children}</div>
  </Container>
)
