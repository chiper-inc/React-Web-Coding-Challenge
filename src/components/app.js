import React from 'react';
import IncidentsList from './incidents_list/incidents_list';
import styled from '../theme';

const Header = styled.header`
  background: #1d3f72;
  color: #fff;
  padding: 20px;
  margin: 0;
  text-align: center;
  border-bottom: 3px solid #71c2cc;

  h1, h2 {
    padding: 0;
    margin: 0;
    display: inline-block;
    vertical-align: middle;
  }

  h1 {
    background: url('logo.png') no-repeat;
    background-size: cover;
    width: 267px;
    height: 100px;
    overflow: hidden;
    text-indent: -999px;
    margin-right: 40px;
  }
`;

function App() {
  return (
    <div className="app">
      <Header>
        <h1>Berlin Polizei</h1>
        <h2>Stolen bykes</h2>
      </Header>
      <IncidentsList></IncidentsList>
    </div>
  );
}

export default App;
