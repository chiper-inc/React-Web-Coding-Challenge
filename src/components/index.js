import BarFilter from './BarFilter';
import Header from './Header';
import BikeItem from './BikeItem';
import Pagination from './Pagination';
import Loading from './Loading';
import ErrorData from './Error';

export { Header, BarFilter, BikeItem, Pagination, Loading, ErrorData };
