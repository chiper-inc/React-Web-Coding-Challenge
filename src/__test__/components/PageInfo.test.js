/* eslint-disable no-undef */
import React from 'react'
import { PageInfo } from '../../components/PageInfo'
import {
  checkRenderComponent,
  checkSnapShot
} from '../checkRenderComponent'

describe('<PageInfo />', () => {
  test('Render of <PageInfo />', () => {
    checkRenderComponent(expect, <PageInfo />)
  })
})

describe('<PageInfo/> Snapshot', () => {
  test('Check UI of OrbitalSpinner', () => {
    checkSnapShot(expect, <PageInfo>test snapshot!</PageInfo>)
  })
})
