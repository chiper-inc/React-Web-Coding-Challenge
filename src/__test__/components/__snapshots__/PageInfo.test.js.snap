// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageInfo/> Snapshot Check UI of OrbitalSpinner 1`] = `
<div
  className="sc-AxjAm kSkoow container row"
>
  <div
    className="col-md-6 align-self-center"
  >
    <img
      src="austronat.svg"
    />
  </div>
  <div
    className="col-md-6 align-self-center"
  >
    test snapshot!
  </div>
</div>
`;
